internal/service/alarm_fetcher.go

```diff
@@ -232,8 +255,157 @@ func (h *AlarmFetcher) doGet(ctx context.Context, path string,
 func (r *AlarmFetcher) mapAlertItem(ctx context.Context,
 	from data.Object) (to data.Object, err error) {
 
-	// TODO: implement mapping
-	to = from
+	alertName, err := data.GetString(from, "labels.alertname")
+	if err != nil {
+		return
+	}
+
+	alarmRaisedTime, err := data.GetString(from, "startsAt")
+	if err != nil {
+		return
+	}
+
+	alarmChangedTime, err := data.GetString(from, "updatedAt")
+	if err != nil {
+		return
+	}
+
+	severity, err := data.GetString(from, "labels.severity")
+	if err != nil {
+		return
+	}
+
+	clusterId, err := data.GetString(from, "labels.managed_cluster")
+	if err != nil {
+		return
+	}
+
+	resourcePool, err := r.FetchResourcePool(ctx, clusterId)
+	if err != nil {
+		return
+	}
+	resourcePoolName, err := data.GetString(resourcePool, "name")
+	if err != nil {
+		return
+	}
+
+	var alarmEventRecordId, resourceID, resourceTypeID string
+	alertInstance, err := data.GetString(from, "labels.instance")
+	if err != nil {
+		// Instance is not available for cluster global alerts
+		alarmEventRecordId = fmt.Sprintf("%s_%s", alertName, resourcePoolName)
+		resourceID = resourcePool["resourcePoolID"].(string)
+	} else {
+		alarmEventRecordId = fmt.Sprintf("%s_%s_%s", alertName, resourcePoolName, alertInstance)
+		resource, err := r.FetchResource(ctx, resourcePoolName, alertInstance)
+		if err == nil {
+			resourceID = resource["resourceID"].(string)
+			resourceTypeID = resource["resourceTypeID"].(string)
+		}
+	}
+
+	// Add the extensions:
+	extensionsMap, err := data.GetExtensions(from, r.extensions, r.jqTool)
+	if err != nil {
+		return
+	}
+	if len(extensionsMap) == 0 {
+		// Fallback to all labels and annotations
+		var labels, annotations data.Object
+		labels, err = data.GetObj(from, "labels")
+		if err != nil {
+			return
+		}
+		annotations, err = data.GetObj(from, "annotations")
+		if err != nil {
+			return
+		}
+		err = mergo.Map(&extensionsMap, labels, mergo.WithOverride)
+		if err != nil {
+			return
+		}
+		err = mergo.Map(&extensionsMap, annotations, mergo.WithOverride)
+		if err != nil {
+			return
+		}
+	}
+
+	to = data.Object{
+		"alarmEventRecordId": alarmEventRecordId,
+		"resourceID":         resourceID,
+		"resourceTypeID":     resourceTypeID,
+		"alarmRaisedTime":    alarmRaisedTime,
+		"alarmChangedTime":   alarmChangedTime,
+		"alarmDefinitionID":  alertName,
+		"probableCauseID":    alertName,
+		"perceivedSeverity":  AlarmSeverity(severity).MapProperty(),
+		"extensions":         extensionsMap,
+	}
+
+	return
+}
+
+func (r *AlarmFetcher) FetchResourcePool(ctx context.Context, clusterId string) (resourcePool data.Object, err error) {
```

Comment:
Do these functions need to be public?

Reply:
Not really, can be private, fixed. Good catch:)